File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/CBuilder/Common/DemoBasicMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DemoBasicMain.h"
#include "DemoRating.h"
#include "DemoBasicAbout.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxLookAndFeels"
#pragma resource "*.dfm"
TDemoBasicMainForm *DemoBasicMainForm;
//---------------------------------------------------------------------------
__fastcall TDemoBasicMainForm::TDemoBasicMainForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actAboutExecute(TObject *Sender)
{
  ShowAbout(true, false);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actDownloadsExecute(TObject *Sender)
{
  ShellExecute(Handle, PAnsiChar("OPEN"), PAnsiChar("http://www.devexpress.com/downloads"), NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actDXOnTheWebExecute(TObject *Sender)
{
  ShellExecute(Handle, PAnsiChar("OPEN"), PAnsiChar("http://www.devexpress.com"), NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actExitExecute(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actForumExecute(TObject *Sender)
{
  ShellExecute(Handle, PAnsiChar("OPEN"), PAnsiChar("https://www.devexpress.com/ClientCenter"), NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actHelpExecute(TObject *Sender)
{
  Application->HelpCommand(HELP_FINDER, 0);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actProductsExecute(TObject *Sender)
{
  ShellExecute(Handle, PAnsiChar("OPEN"), PAnsiChar("http://www.devexpress.com/products"), NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actRateDemoExecute(TObject *Sender)
{
  TDemoRatingForm *AForm = new TDemoRatingForm(this);
    __try{
      AForm->ShowModal();
    }
    __finally{
      AForm->Free();
    }
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::actShowDemoDescriptionExecute(TObject *Sender)
{
  lscrip->Visible = !lscrip->Visible;
  ((TCustomAction*)Sender)->Checked = !((TCustomAction*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void TDemoBasicMainForm::AdjustAboutText(TStrings *AAboutText)
{
  AAboutText->Assign(memAboutText->Lines);
}
//---------------------------------------------------------------------------

void TDemoBasicMainForm::ShowAbout(bool AModal, bool AOnTop)
{
  if (DemoBasicAboutForm == NULL)
    DemoBasicAboutForm = new TDemoBasicAboutForm(Application);
  AdjustAboutText(DemoBasicAboutForm->reDemoInfo->Lines);
  DemoBasicAboutForm->lmoName->Caption = ChangeFileExt(ExtractFileName(Application->ExeName),"");
  if (AOnTop)
    DemoBasicAboutForm->FormStyle = fsStayOnTop;
  else
    DemoBasicAboutForm->FormStyle = fsNormal;
  if (AModal)
    DemoBasicAboutForm->ShowModal();
  else
    DemoBasicAboutForm->Show();
}
//---------------------------------------------------------------------------


void __fastcall TDemoBasicMainForm::actSupportExecute(TObject *Sender)
{
  ShellExecute(Handle, PAnsiChar("OPEN"), PAnsiChar("http://www.devexpress.com/Support/Center"), NULL, NULL, SW_SHOWMAXIMIZED);
}
//---------------------------------------------------------------------------

File: DevExpress VCL/ExpressQuantumTreeList 4/Demos/Delphi/Common/DemoBasicMain.pas
unit DemoBasicMain;

{$I ..\cxVer.inc}

interface

uses
  Windows, Messages, SysUtils{$IFDEF DELPHI6}, Variants{$ENDIF}, Classes, Graphics, Controls, Forms,
  Dialogs, cxControls, cxLookAndFeels, ActnList, ImgList, Menus, ComCtrls,
  StdCtrls;

type
  TDemoBasicMainForm = class(TForm)
    mmMain: TMainMenu;
    miAbout: TMenuItem;
    miHelp: TMenuItem;
    miDeveloperExpressontheweb: TMenuItem;
    miForum: TMenuItem;
    miDownloads: TMenuItem;
    miSeparator4: TMenuItem;
    miGridHelp: TMenuItem;
    miProducts: TMenuItem;
    miSeparator3: TMenuItem;
    ilMain: TImageList;
    alMain: TActionList;
    actHelp: TAction;
    actProducts: TAction;
    actDownloads: TAction;
    actForum: TAction;
    actDXOnTheWeb: TAction;
    actAbout: TAction;
    miFile: TMenuItem;
    miOptions: TMenuItem;
    miExit: TMenuItem;
    actExit: TAction;
    sbMain: TStatusBar;
    lscrip: TLabel;
    actShowDemoDescription: TAction;
    miShowDemoDescription: TMenuItem;
    cxLookAndFeelController: TcxLookAndFeelController;
    miRate: TMenuItem;
    actRateDemo: TAction;
    memAboutText: TMemo;
    actSupport: TAction;
    miSupport: TMenuItem;
    procedure actProductsExecute(Sender: TObject);
    procedure actDownloadsExecute(Sender: TObject);
    procedure actForumExecute(Sender: TObject);
    procedure actDXOnTheWebExecute(Sender: TObject);
    procedure actExitExecute(Sender: TObject);
    procedure actShowDemoDescriptionExecute(Sender: TObject);
    procedure actHelpExecute(Sender: TObject);
    procedure actRateDemoExecute(Sender: TObject);
    procedure actAboutExecute(Sender: TObject);
    procedure actSupportExecute(Sender: TObject);
  protected
    procedure AdjustAboutText(AAboutText: TStrings); virtual;
    procedure ShowAbout(AModal, AOnTop: Boolean); virtual;
  end;

var
  DemoBasicMainForm: TDemoBasicMainForm;

implementation

uses DemoBasicAbout, ShellAPI, DemoRating;

{$R *.dfm}

procedure TDemoBasicMainForm.actProductsExecute(Sender: TObject);
begin
  ShellExecute(Handle, PChar('OPEN'), PChar('http://www.devexpress.com/products'), nil, nil, SW_SHOWMAXIMIZED);
end;

procedure TDemoBasicMainForm.actDownloadsExecute(Sender: TObject);
begin
  ShellExecute(Handle, PChar('OPEN'), PChar('http://www.devexpress.com/downloads'), nil, nil, SW_SHOWMAXIMIZED);
end;

procedure TDemoBasicMainForm.actForumExecute(Sender: TObject);
begin
  ShellExecute(Handle, PChar('OPEN'), PChar('https://www.devexpress.com/ClientCenter'), nil, nil, SW_SHOWMAXIMIZED);
end;

procedure TDemoBasicMainForm.actDXOnTheWebExecute(
  Sender: TObject);
begin
  ShellExecute(Handle, PChar('OPEN'), PChar('http://www.devexpress.com'), nil, nil, SW_SHOWMAXIMIZED);
end;

procedure TDemoBasicMainForm.actExitExecute(Sender: TObject);
begin
  Close;
end;

procedure TDemoBasicMainForm.actShowDemoDescriptionExecute(
  Sender: TObject);
begin
  lscrip.Visible := not lscrip.Visible;
  TCustomAction(Sender).Checked := not TCustomAction(Sender).Checked;
end;

procedure TDemoBasicMainForm.actHelpExecute(Sender: TObject);
begin
  Application.HelpCommand(HELP_FINDER, 0);
end;

procedure TDemoBasicMainForm.AdjustAboutText(AAboutText: TStrings);
begin
  AAboutText.Assign(memAboutText.Lines);
end;

procedure TDemoBasicMainForm.actRateDemoExecute(Sender: TObject);
begin
  with TDemoRatingForm.Create(Self) do
    try
      ShowModal;
    finally
      Free;
    end;
end;

procedure TDemoBasicMainForm.ShowAbout(AModal, AOnTop: Boolean);
begin
  if not Assigned(DemoBasicAboutForm) then
    DemoBasicAboutForm := TDemoBasicAboutForm.Create(Application);
  with DemoBasicAboutForm do
  begin
    AdjustAboutText(reDemoInfo.Lines);
    lmoName.Caption := ChangeFileExt(ExtractFileName(Application.ExeName),'');
    if AOnTop then
      FormStyle := fsStayOnTop
    else
      FormStyle := fsNormal;
    if AModal then
      ShowModal
    else
      Show;
  end;
end;

procedure TDemoBasicMainForm.actAboutExecute(Sender: TObject);
begin
  ShowAbout(True, False);
end;

procedure TDemoBasicMainForm.actSupportExecute(Sender: TObject);
begin
  ShellExecute(Handle, PChar('OPEN'), PChar('http://www.devexpress.com/Support/Center'), nil, nil, SW_SHOWMAXIMIZED);
end;

end.

